.circleci/cimodel/data/binary_build_definitions.py

```diff
@@ -85,7 +86,7 @@ def gen_workflow_job(self, phase, upload_phase_dependency=None, nightly=False):
             job_def["libtorch_variant"] = miniutils.quote(self.libtorch_variant)
         if phase == "test":
             if not self.smoke:
-                job_def["requires"].append(self.gen_build_name("build", nightly))
+                job_def["requires"] = [self.gen_build_name("build", nightly)]
```

Comment:
Why did this stop being append?

Reply:
because I didn't want `requires` to be filled out unless it was actually needed

Reply:
The original line for `jobdef["requires"]` was `jobdef["requires"] = ["setup"]`

Reply:
In particular, this line is not obviously mutually exclusive with https://github.com/pytorch/pytorch/pull/39081/files#diff-8cc86e31d716e5f0a0c9883a5112d700R68

Reply:
It should be considering the conditionals are `if self.smoke` and `if not self.smoke`

Reply:
I don't disagree that it creates a bit of confusion, but previously I couldn't get the YAML outputter to work nicely with the empty lists for requires.

Reply:
NP. I verified that the generated code looks fine.